aws/resource_aws_vpc_peering_connection_options.go

```diff
@@ -53,25 +54,37 @@ func resourceAwsVpcPeeringConnectionOptionsRead(d *schema.ResourceData, meta int
 
 	d.Set("vpc_peering_connection_id", pc.VpcPeeringConnectionId)
 
-	if pc != nil && pc.AccepterVpcInfo != nil && pc.AccepterVpcInfo.PeeringOptions != nil {
-		err := d.Set("accepter", flattenVpcPeeringConnectionOptions(pc.AccepterVpcInfo.PeeringOptions))
-		if err != nil {
-			return fmt.Errorf("Error setting VPC Peering Connection Options accepter information: %s", err.Error())
-		}
+	if err := d.Set("accepter", flattenVpcPeeringConnectionOptions(pc.AccepterVpcInfo.PeeringOptions)); err != nil {
+		return fmt.Errorf("Error setting VPC Peering Connection Options accepter information: %s", err)
 	}
-
-	if pc != nil && pc.RequesterVpcInfo != nil && pc.RequesterVpcInfo.PeeringOptions != nil {
-		err := d.Set("requester", flattenVpcPeeringConnectionOptions(pc.RequesterVpcInfo.PeeringOptions))
-		if err != nil {
-			return fmt.Errorf("Error setting VPC Peering Connection Options requester information: %s", err.Error())
-		}
+	if err := d.Set("requester", flattenVpcPeeringConnectionOptions(pc.RequesterVpcInfo.PeeringOptions)); err != nil {
+		return fmt.Errorf("Error setting VPC Peering Connection Options requester information: %s", err)
 	}
 
 	return nil
 }
 
 func resourceAwsVpcPeeringConnectionOptionsUpdate(d *schema.ResourceData, meta interface{}) error {
-	if err := resourceAwsVpcPeeringConnectionModifyOptions(d, meta); err != nil {
+	conn := meta.(*AWSClient).ec2conn
```

Comment:
Need to get the peering connection info here to determine whether or not it's a cross-region peering.

Reply:
I think in the future we can avoid logic like this by saving this information into a computed attribute.